### ISSUE TITLE
Your yaml is invalid

### ISSUE THREAD

**Issue by TheAngryByrd** [MattHodge/PackerTemplates#13]
```
System.IO.FileNotFoundException: Could not load file or assembly 'YamlDotNet, Version=4.2.1.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies.
File name: 'YamlDotNet, Version=4.2.1.0, Culture=neutral, PublicKeyToken=null'
  at Submission#0.DeserializeYamlFromFile[T] (Cake.Core.IO.FilePath filename) [0x0000c] in <cb8a275aead04fb8b7e8455004212668>:0 
  at Submission#0.LoadYAMLConfig (System.String yaml_path, System.String os) [0x00016] in <cb8a275aead04fb8b7e8455004212668>:0 
Error: One or more errors occurred.
	Your yaml is invalid.
```

**Comment by johnrizzo1** [comment]
same issue here.  You come up with a fix?

**Comment by TheAngryByrd** [comment]
Yeah I added YamlDotNet as a reference to cake then modified the packages.config under tools to pin to 4.2.1

Cake should use paket 😊 but that's another topic.

**Comment by johnrizzo1** [comment]
Thank you.  I've basically merged your changes as well as those in Pull Request 10 and 12.  I'd love to continue to contribute to the project as well.  I'm going to continue to develop the project in my fork for now.  johnrizzo1/PackerTemplates.  I'll push my work shortly.

**Comment by johnrizzo1** [comment]
@TheAngryByrd can you post your build.supported_os.yaml?

**Comment by TheAngryByrd** [comment]
It's just the yaml from the initial clone. 


```
---
- Name: Windows2012R2Core
  osName: win2012r2core
  guestOSType: Windows2012_64
  isoURL: http://care.dlservice.microsoft.com/dl/download/6/2/A/62A76ABB-9990-4EFC-A4FE-C7D698DAEB96/9600.17050.WINBLUE_REFRESH.140317-1640_X64FRE_SERVER_EVAL_EN-US-IR3_SSS_X64FREE_EN-US_DV9.ISO
  isoChecksum: 849734f37346385dac2c101e4aacba4626bb141c
- Name: Windows2012R2
  osName: win2012r2
  guestOSType: Windows2012_64
  isoChecksum: 849734f37346385dac2c101e4aacba4626bb141c
  isoURL: http://care.dlservice.microsoft.com/dl/download/6/2/A/62A76ABB-9990-4EFC-A4FE-C7D698DAEB96/9600.17050.WINBLUE_REFRESH.140317-1640_X64FRE_SERVER_EVAL_EN-US-IR3_SSS_X64FREE_EN-US_DV9.ISO
- Name: Windows2016StdCore
  osName: win2016stdcore
  guestOSType: Windows2012_64
  isoChecksum: 3bb1c60417e9aeb3f4ce0eb02189c0c84a1c6691
  isoURL: http://care.dlservice.microsoft.com/dl/download/1/6/F/16FA20E6-4662-482A-920B-1A45CF5AAE3C/14393.0.160715-1616.RS1_RELEASE_SERVER_EVAL_X64FRE_EN-US.ISO
- Name: Windows2016
  osName: win2016std
  guestOSType: Windows2012_64
  isoChecksum: 3bb1c60417e9aeb3f4ce0eb02189c0c84a1c6691
  isoURL: http://care.dlservice.microsoft.com/dl/download/1/6/F/16FA20E6-4662-482A-920B-1A45CF5AAE3C/14393.0.160715-1616.RS1_RELEASE_SERVER_EVAL_X64FRE_EN-US.ISO
# - Name: Windows10
#   osName: win10
#   guestOSType: Windows10_64
#   isoChecksum: 56ab095075be28a90bc0b510835280975c6bb2ce
#   isoURL: http://care.dlservice.microsoft.com/dl/download/C/3/9/C399EEA8-135D-4207-92C9-6AAB3259F6EF/10240.16384.150709-1700.TH1_CLIENTENTERPRISEEVAL_OEMRET_X64FRE_EN-US.ISO

```

**Comment by johnrizzo1** [comment]
Odd.  Mine is the same but I get this error...

❯ ./build.ps1
Preparing to run build script...
Running build script...
Analyzing build script...
Processing build script...
Installing addins...
Compiling build script...
Error: One or more errors occurred.
        Your yaml is invalid.

I added in some debugging and I get this output

YAML path ./build.supported_os.yaml OS Windows2016StdCore - Matched OS

Here are my changes to add some debugging.

```cake
public OSToBuild LoadYAMLConfig(string yaml_path, string os)
{
    //OSToBuild os_to_build_properties;
    string matchedOs = "";

    try
    {
        var oses = DeserializeYamlFromFile<List<OSToBuild>> (yaml_path);

        // check if the OS the user passed exists
        bool matchingOS = oses.Any(n => n.Name == os);
        Information("oses: " + oses);
        if (matchingOS == true)
        {
            // return the matching os to build
            return oses.Where(n => n.Name == os).First();
        }
        else
        {
            string exceptionMsg = String.Format("Could not find a matching operating system in {0}. You passed in: {1}", yaml_path, os);
            throw new System.ArgumentException(exceptionMsg);
        }
    }
    catch
    {
      string msg = String.Format("Your yaml is invalid. YAML path {0} OS {1} - Matched OS {2}", yaml_path, os, matchedOs);
      throw new System.ArgumentException(msg);
    }
}
```

**Comment by TheAngryByrd** [comment]
so I added to `build.cake` 
`
#addin "YamlDotNet"
`
and to `tools/packages.config`
`
    <package id="YamlDotNet" version="4.2.1" />
`

Seems to give work around.  Stupid binding redirect nonsense.

**Comment by johnrizzo1** [comment]
Thanks... That worked.  Again, I'll add this to my repo so that it is all captured somewhere.

**Comment by MattHodge** [comment]
I have added this fix (8ca65c4) to the repo, thanks @TheAngryByrd
